src/languageservice/services/yamlCompletion.ts

```diff
@@ -338,6 +339,16 @@ export class YamlCompletion {
     let overwriteRange: Range = null;
     if (areOnlySpacesAfterPosition) {
       overwriteRange = Range.create(position, Position.create(position.line, lineContent.length));
+      if (node && isScalar(node) && lineContent.match(/\s+$/)) {
+        // line contains part of a key with trailing spaces, adjust the overwrite range to include only the text
+        const matches = lineContent.match(/^(\s*)(?!\s*-)(?!\s*\w+:)(?!\s+$)(?!$)/);
```

Comment:
super cool regexp it took me a while to understand it :)
I noticed that you exclude `-` from mach
I tried it and the issue is the same with the array value
<img width="175" alt="image" src="https://user-images.githubusercontent.com/38421337/176199689-a5b98f74-5c60-4fd4-a38b-a3f018a34476.png">
is it possible to modify the regexp to allow `-`, probably multiple times
` - - key  ` is also valid yaml